src/main/java/hazell/Ui.java

```diff
@@ -0,0 +1,46 @@
+package hazell;
+
+import java.util.Scanner;
+
+/**
+ * A class that abstracts the interactions with user.
+ */
+public class Ui {
+    Scanner scanner;
+
+    /**
+     * Create a new Ui instance.
+     */
+    public Ui() {
+        this.scanner = new Scanner(System.in);
+    }
+
+    /**
+     * Check if user has entered a new command.
+     * @return Whether user has entered a new command
+     */
+    public boolean hasNextCommand() {
+        return this.scanner.hasNextLine();
+    }
```

Comment:
Maybe can consider rephrasing comments to match java coding standard (i.e. CHECKS if user has entered a new commmand.)